Update: Add true curves APIs to `Add graphics with renderer` (#1089)
Esri/arcgis-runtime-samples-ios: arcgis-ios-sdk-samples/Display information/Add graphics with renderer/README.md

## How to use the sample

> Run the sample and view graphics for points, lines, polygons, and shapes with curved segments which are stylized using renderers.

Reviewer: ```suggestion
Pan and zoom on the map to view graphics for points, lines, polygons, and shapes with curved segments which are stylized using renderers.
```
I think it's been a while since we've updated this sample and README. A slight change would keep this up to date with our current guidelines.

> ## How it works

Reviewer: I didn't notice this before since no changes were made to this section, but the list should be enumerated. I also wonder if some explanation should be added to go into depth about the added API. Seems a bit odd to have a long list of Relevant API but disregard half of them in this section.

Author: Do you mean to change it to ...?
> Pan and zoom on the map to view graphics for points, lines, **and** polygons (including polygons with curve segments) which are stylized using renderers.

> I also wonder if some explanation should be added to go into depth about the added API. Seems a bit odd to have a long list of Relevant API but disregard half of them in this section.

It would have been a good question to ask in the design update PR. 

IMO though it is OK to omit them from the `How it works` section. This sample is intended for exhibiting how to apply renderers to graphics overlays, not how to construct geometries. True curve APIs are only included in the code so that user can find example snippet when they search for these keywords. @TADraeseke may chime in here.

Reviewer: I think others may have missed this section in the design update as well since this section wasn't highlighted in either PRs. Feel free to request Phil's and Sarat's reviews so they can chime in as well

Reviewer 2: Oh yeah! Should be enumerated according to our [README template](https://devtopia.esri.com/runtime/common-samples/wiki/README-Template#how-it-works). 

As to Relevant API. Really good point @vquach2404. This sample's main purpose is to show use of a renderer and it's been co-opted to show a curve since it's an easy win until we build out more functionality next release.

Let's not worry about changing the steps too much now (though they should be enumerated). This sample will serve as a basic story that we support curves, but we'll go into detail on how in another sample next release
